pkg/operator/cockroachdb/controller.go

```diff
@@ -332,22 +332,7 @@ func (c *ClusterController) createStatefulSet(cluster *cluster) error {
 			UpdateStrategy: appsv1beta1.StatefulSetUpdateStrategy{
 				Type: appsv1beta1.RollingUpdateStatefulSetStrategyType,
 			},
-			VolumeClaimTemplates: []v1.PersistentVolumeClaim{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      volumeNameDataDir,
-						Namespace: cluster.namespace,
-					},
-					Spec: v1.PersistentVolumeClaimSpec{
-						AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
-						Resources: v1.ResourceRequirements{
-							Requests: v1.ResourceList{
-								v1.ResourceStorage: cluster.spec.VolumeSize,
```

Comment:
looks like this has migration/upgrade implications.  If someone still has a cockroach CRD that is specifying `volumeSize` of x GB, what is going to happen?  Maybe nothing will happen since the stateful set has already been created and this `createStatefulSet` function does not call Update to update the existing stateful set.  We should think through this and test this.

we didn't remove `VolumeSize` from the spec in types.go, so should we continue honoring that value? (create a volume claim template of that size).  Should that field be completely removed from the spec?

Reply:
@jbw976 I need to look into how I can make a good migration path for Minio and CockroachDB there.

Right now I would want to remove the `volumeSize` completely and go with `volumeClaimTemplates` only.

What do you think?

Reply:
i thought through this a little more.  for existing deployments, the statefulset will already exist and since we don't support updates, it will just keep running as is and be OK.

for new deployments, as long as we've updated the documentation and examples to remove volumeSize, then we no longer have to support it at all.  i feel OK with this because existing scenarios will not just all of a sudden break.  they should continue running OK.

do you also think that existing users will not break @galexrt?